### PR TITLE
docs: Add build script to test templates

### PR DESCRIPTION
**Pull request by sdankel**
## Description

Closes https://github.com/FuelLabs/sway/issues/3026

Adds a build script to the test templates that runs prior to the tests when `cargo test` is run. This makes it so users who copy this template don't have to run `forc build` prior to `cargo test`

## Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: templates/sway-predicate-test-rs/template/build.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+//! This build script is used to compile the sway project using the `forc` prior to running tests.
```
Comment (K1-R1) [inline templates/sway-predicate-test-rs/template/build.rs:1]:
```suggestion
//! This build script is used to compile the sway project using `forc` prior to running tests.
```

File: templates/sway-script-test-rs/template/build.rs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+//! This build script is used to compile the sway project using the `forc` prior to running tests.
```
Comment (K1-R1) [inline templates/sway-script-test-rs/template/build.rs:1]:
```suggestion
//! This build script is used to compile the sway project using `forc` prior to running tests.
```

File: templates/sway-test-rs/template/build.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+//! This build script is used to compile the sway project using the `forc` prior to running tests.
```
Comment (K1-R1) [inline templates/sway-test-rs/template/build.rs:1]:
```suggestion
//! This build script is used to compile the sway project using `forc` prior to running tests.
```

**Review by K1-R1 — COMMENTED** [review]
This is a really nice addition, just a few nits

**Review by K1-R1 — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

**Review by alfiedotwtf — APPROVED** [review]

**Pull request merged**